Decimation/Simplify.h

```diff
@@ -87,942 +83,977 @@ struct vec3f
+            //
+            // All triangles with edges below the threshold will be removed
+            //
+            // The following numbers works well for most models.
+            // If it does not, try to adjust the 3 parameters
+            //
+            double threshold = DBL_EPSILON; //1.0E-3 EPS;
+            if (verbose) {
+                printf("lossless iteration %d\n", iteration);
+            }
+
+            // remove vertices & mark deleted triangles
+            for(Triangle& t: triangles)
+            {
+                if(t.err[3]>threshold) continue;
+                if(t.deleted) continue;
+                if(t.dirty) continue;
+
+                for(size_t j: {0, 1, 2})
+                {
+                    if(t.err[j] >= threshold)
+                    {
+                        continue;
+                    }
+                    int i0=t.v[ j     ]; Vertex &v0 = vertices[i0];
+                    int i1=t.v[(j+1)%3]; Vertex &v1 = vertices[i1];
+
+                    // Border check
+                    if(v0.border != v1.border)  continue;
+
+                    // Compute vertex to collapse to
+                    vec3f p;
+                    calculate_error(i0,i1,p);
+
+                    deleted0.resize(v0.tcount); // normals temporarily
+                    deleted1.resize(v1.tcount); // normals temporarily
+
+                    // don't remove if flipped
+                    if( flipped(p,i0,i1,v0,v1,deleted0) ) continue;
+                    if( flipped(p,i1,i0,v1,v0,deleted1) ) continue;
+
+                    if ( (t.attr & TEXCOORD) == TEXCOORD )
+                    {
+                        update_uvs(i0,v0,p,deleted0);
+                        update_uvs(i0,v1,p,deleted1);
+                    }
+
+                    // not flipped, so remove edge
+                    v0.p=p;
+                    v0.q=v1.q+v0.q;
+                    size_t tstart = refs.size();
+
+                    update_triangles(i0,v0,deleted0,deleted_triangles);
+                    update_triangles(i0,v1,deleted1,deleted_triangles);
+
+                    size_t tcount = refs.size() - tstart;
+
+                    if(tcount <= v0.tcount)
+                    {
+                        // save ram
+                        if(tcount)memcpy(&refs[v0.tstart],&refs[tstart],tcount*sizeof(Ref));
+                    }
+                    else
+                        // append
+                        v0.tstart=tstart;
+
+                    v0.tcount=tcount;
+                    break;
+                }
+            }
+            if(deleted_triangles<=0)break;
+            deleted_triangles=0;
+        } //for each iteration
+        // clean up mesh
+        compact_mesh();
+    } //simplify_mesh_lossless()
+
+
+    // Check if a triangle flips when this edge is removed
+
+    bool flipped(vec3f p,int i0,int i1,Vertex &v0,Vertex &v1,std::vector<int> &deleted)
+    {
+        (void)i0; // unused
+        (void)v1; // unused
```

Comment:
Can we just delete these variables?

Reply:
This is too avoid the unused parameter warnings.

I initially wanted to avoid modifying too much of the code, I will do this a follow up PR